auth0/src/test/java/com/auth0/android/authentication/storage/SecureCredentialsManagerTest.java

```diff
@@ -750,6 +751,36 @@ public void shouldGetCredentialsAfterAuthentication() {
         assertThat(retryCheck, is(false));
     }
 
+    @Test
+    public void shouldNotGetCredentialsWhenCredentialsHaveExpired() {
+        Date credentialsExpiresAt = new Date(CredentialsMock.CURRENT_TIME_MS + 123456L * 1000);
+        Date storedExpiresAt = new Date(CredentialsMock.CURRENT_TIME_MS - 60L * 1000);
+        insertTestCredentials(true, true, false, credentialsExpiresAt);
+        when(storage.retrieveLong("com.auth0.credentials_expires_at")).thenReturn(storedExpiresAt.getTime());
+
+        //Require authentication
+        Activity activity = spy(Robolectric.buildActivity(Activity.class).create().start().resume().get());
+        KeyguardManager kService = mock(KeyguardManager.class);
+        when(activity.getSystemService(Context.KEYGUARD_SERVICE)).thenReturn(kService);
+        when(kService.isKeyguardSecure()).thenReturn(true);
+        Intent confirmCredentialsIntent = mock(Intent.class);
+        when(kService.createConfirmDeviceCredentialIntent("theTitle", "theDescription")).thenReturn(confirmCredentialsIntent);
+        boolean willRequireAuthentication = manager.requireAuthentication(activity, 123, "theTitle", "theDescription");
+        assertThat(willRequireAuthentication, is(true));
+
+        manager.getCredentials(callback);
+
+        //Should fail because of expired credentials
+        verify(callback).onFailure(exceptionCaptor.capture());
+        CredentialsManagerException exception = exceptionCaptor.getValue();
+        assertThat(exception, is(notNullValue()));
+        assertThat(exception.getMessage(), is("No Credentials were previously set."));
+
+        //A second call to checkAuthenticationResult should fail as callback is set to null
+        final boolean retryCheck = manager.checkAuthenticationResult(123, Activity.RESULT_OK);
```

Comment:
Should this be in this test?

Reply:
Yes. One of the assumptions is that this method cannot be called without previously having started the authentication. This checks that any failure scenario, including this particular one, does invalidate any further call until the whole flow it's started again.